default.nix

```diff
@@ -2,6 +2,7 @@
 # the names here can clash with non-haskell dependencies
 # make sure to specify the names the explicitly that may be clashing
 { callPackage
+, mkDerivation
```

Comment:
What's this for?

Reply:
`staticJustExecutable` tries to pass in `mkDerivation`. 
It uses `overrideCabal` internally which overrides the `mkDerivation`.

https://github.com/NixOS/nixpkgs/blob/046c6a7038998507134981c8a294bb70861484bf/pkgs/development/haskell-modules/lib.nix#L37